File: Assets/GraphQL/Model/Users/Guide.cs
using GraphQL.Helpers;

namespace GraphQL.Model {
	public class Guide {
		[GqlSelection("guideID")] public int GuideID { get; private set; }
		[GqlSelection("fName")] public string FName { get; private set; }
		[GqlSelection("eMail")] public string EMail { get; private set; }
		[GqlSelection("passwordKey")] public string PasswordKey { get; private set; }
	}
}

File: Server/src/schema/Users/guide.ts
import { Field, InputType, Int, ObjectType } from "type-graphql";
import { BaseEntity, Column, Entity, PrimaryGeneratedColumn } from "typeorm";
import BaseUser from "./baseUser";

@Entity()
@ObjectType()
export default class Guide extends BaseUser {
    @Field(() => Int)
    override type: number = this.getType();

    @Field(() => Int!)
    @PrimaryGeneratedColumn()
    guideID!: number;

    @Field(() => String)
    @Column({default: ""})
    phone: string = "";

    setID(): void {
        this.ID = this.guideID;
    }

    getType(): number {
        return 1;
    }

    constructor() {
        super();

        this.guideID = 0;
        this.fName = "None";
        this.eMail = "None@email.com";
        this.passwordKey = "None";
    }
}